src/sql-string.js

```diff
@@ -76,9 +77,54 @@ function escape(val, timeZone, dialect, format) {
       val = val.replace(/\0/g, '\\0');
     }
   } else if (dialect === 'oracle' && typeof val === 'string') {
-    if (val.startsWith('TO_TIMESTAMP') || val.startsWith('TO_DATE')) {
+    if (val.startsWith('TO_TIMESTAMP_TZ') || val.startsWith('TO_DATE')) {
+      const splitVal = val.split(/\(|\)/);
+    
+      if (splitVal.length !== 3 || splitVal[2] !== '') {
```

Comment:
If I understand this right, we are checking here for the fact that we get something like `TO_DATE(col)` and have nothing coming after the closing paren? Please add a comment explaining this

Reply:
Yes, comment added